### PR TITLE
feat(api-graphql): alias authMode identityPool -> iam

### PR DESCRIPTION
**Pull request by iartemiev**
#### Description of changes

* support `authMode: 'identityPool'`, which is an alias for `'iam'`

#### Description of how you validated changes
* added unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/singleton/API/types.ts [hunk 1]
```diff
@@ -82,6 +82,7 @@ export type GraphQLAuthMode =
 	| 'oidc'
 	| 'userPool'
 	| 'iam'
```
Comment (HuiSF) [inline packages/core/src/singleton/API/types.ts:84]: Should we start marking `iam` as `@deprecated` so we can remove it from the next major version?

**Review by svidgen — APPROVED** [review]

File: packages/core/src/singleton/API/types.ts (hunk 1 above)
Comment (iartemiev) [reply packages/core/src/singleton/API/types.ts:84]: I tried that, but it doesn't actually surface over union types AFAICT

Comment (iartemiev) [reply packages/core/src/singleton/API/types.ts:84]: Maybe still worth it as a reminder to ourselves. I'll update

Comment (iartemiev) [reply packages/core/src/singleton/API/types.ts:84]: Screenshot 2024-04-26 at 2 14 30 PM

**Review by svidgen — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Pull request merged**